airbnb/knowledge-repo
# Automatic code folding for improved readability

Hey there,

Here's a feature that could prove useful to everyone: I've added a javascript snippet to the rendered markdown template that pins 'Hide' or 'Expand' buttons in the top right corner of every code chunk (and a 'Hide All Code' / 'Show All Code' button on top of the page). I saw there was an open ticket for it as well: #118

By default every code chunk is shown, but it crops-scroll those taller than 400px.

Here's how it looks like:

- [http://i.imgur.com/JBZev6c.png](http://i.imgur.com/JBZev6c.png)
- [http://i.imgur.com/K0ViJXp.png](http://i.imgur.com/K0ViJXp.png)

(Everything is orange because we customized it to our theme but the buttons should be colored with the regular blue-green in the pull request)

**Test Plan:** 

1. Run a local instance of the knowledge_repo
2. Check out or preview any post containing at least one code chunk

Auto-reviewers: @NiharikaRay @matthewwardrop @earthmancash @danfrankj
